### DISCUSSION TITLE
JavaScript not separated by semicolon in bundle

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [datenstrom/community#530] [Report a bug] [answered]
I stumbled upon a little nasty problem in bundle.php: when a script begins with an open parenthesis, the last token of the previous bundled script is interpreted by the Javascript parser as a function (and nothing works, obviously). It seems to me that all is solved by adding a final semicolon with the line 119, like this:

```php
if (!empty($fileDataBundle)) $fileDataBundle .= ";\n\n";
```

Am I right?

**Comment by markseuffert** [comment]
I can confirm this problem, we stumbled across it a few weeks ago in datenstrom/community#514.

**Comment by markseuffert** [comment, marked as answer]
A new bundle extension is available. As you suggested, a semicolon separator is added between bundled JavaScript files. I moved the implementation into `class MinifyJavaScript` so that it doesn't affect CSS.

Should be working now. Does that fix your problem?

**Comment by GiovanniSalmeri** [comment]
All is perfect, thank you!
